src/lib/hooks/swap/index.ts

```diff
@@ -0,0 +1,63 @@
+import { Currency } from '@uniswap/sdk-core'
+import { useAtom } from 'jotai'
+import { useAtomValue, useUpdateAtom } from 'jotai/utils'
+import { pickAtom } from 'lib/state/atoms'
+import { amountAtom, Field, independentFieldAtom, swapAtom } from 'lib/state/swap'
+import { useCallback, useMemo } from 'react'
+export { default as useSwapInfo } from './useSwapInfo'
+
+function useCurrency(field: Field): [Currency | undefined, (currency?: Currency) => void] {
+  const atom = useMemo(() => pickAtom(swapAtom, field), [field])
+  return useAtom(atom)
+}
+
+export function useInputCurrency() {
```

Comment:
Considering that `Input.tsx`/`Output.tsx` use `Field.INPUT`/`Field.OUTPUT` already, would it be simpler to just export `useCurrency`/`useAmount` and not bother with field-specific named exports?

Reply:
yeah i personally like useCurrency and passing in the field - less hooks and code overall

Reply:
i get a weird feeling in the current setup

Reply:
1. we already have a `useCurrency` (`src\lib\hooks\useCurrency.ts`) w/ a diff signature and use. 

2. this will turn into us doing the following a lot, instead, right?

```
const inputCurrency = useCurrency(Field.INPUT)
const outputCurrency = useCurrency(Field.OUTPUT)
```

Reply:
true having the same hook name would feel strange

Reply:
we could call it `useSwapCurrency` - would that resolve it?

Reply:
I think I'd choose `useSwapFieldCurrency`, but yeah that works for sure. 

Doesn't it amount to the same thing, though? We will be using the same logic and afaik it doesn't change how the hook will be used.

Reply:
kewl, I'm fine with it!